# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../make/make-3.82-0002-oneshell.patch
# Copyright (C) 2012 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

From cfa19c5629f0252c4ab9058cadb582890ec3300b Mon Sep 17 00:00:00 2001
From: Christian Wiese <chris@opensde.org>
Date: Mon, 22 Oct 2012 17:47:09 +0200
Subject: [PATCH] oneshell

ChangeLog:

2010-08-13  Paul Smith  <psmith@gnu.org>

       * main.c (main): Add "oneshell" to $(.FEATURES) (forgot to add
       this in 3.82!)

test/ChangeLog:

       * run_make_tests.pl (set_more_defaults): Set a %FEATURES hash to
       the features available in $(.FEATURES).
---
 main.c                  |    2 +-
 tests/run_make_tests.pl |    3 +++
 2 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/main.c b/main.c
index 9fe8090..782b0de 100644
--- a/main.c
+++ b/main.c
@@ -1138,7 +1138,7 @@ main (int argc, char **argv, char **envp)
      a macro and some compilers (MSVC) don't like conditionals in macros.  */
   {
     const char *features = "target-specific order-only second-expansion"
-                           " else-if shortest-stem undefine"
+                           " else-if shortest-stem undefine oneshell"
 #ifndef NO_ARCHIVES
                            " archives"
 #endif
diff --git a/tests/run_make_tests.pl b/tests/run_make_tests.pl
index 2c8c08b..7291c55 100755
--- a/tests/run_make_tests.pl
+++ b/tests/run_make_tests.pl
@@ -29,6 +29,7 @@
 # You should have received a copy of the GNU General Public License along with
 # this program.  If not, see <http://www.gnu.org/licenses/>.
 
+%FEATURES = ();
 
 $valgrind = 0;              # invoke make with valgrind
 $valgrind_args = '';
@@ -367,6 +368,8 @@ sub set_more_defaults
      $parallel_jobs = 1;
    }
 
+   %FEATURES = map { $_ => 1 } split /\s+/, `sh -c "echo '\\\$(info \\\$(.FEATURES))' | $make_path -f- 2>/dev/null"`;
+
    # Set up for valgrind, if requested.
 
    if ($valgrind) {
-- 
1.7.2.3

